blog/2022-11-09-apache-monitoring.md

```diff
@@ -0,0 +1,144 @@
+
+![image](https://user-images.githubusercontent.com/24860547/200813688-67011d91-2fd1-48ee-a492-bb7409609e06.png)
+
+### Workers
+
+Worker resources are important to monitor as this tells you which resources are over and under utilized. 
+
+The worker threads chart represents worker utilization. 
+
+A worker thread that is in any of the following states is considered busy: 
+- reading
+- writing
+- keep-alive
+- logging
+- closing
+- gracefully finishing
+
+A worker thread not in any of the states mentioned above is considered idle. 
+
+A consistently large number of idle workers (as seen in the example chart here) indicates that more threads are in use than are necessary for the current traffic levels and load. This will lead to unnecessary utilization of system resources and you may consider lowering the `MinSpareThreads` configuration parameter.
+
+If however you only have a very small number of idle workers consistently this could lead to slowing down your server and requests getting queued up when the `MaxRequestWorkers` limit is hit. Increasing the `MaxRequestWorkers` can help with this scenario but be mindful that each extra worker thread requires extra system resources. 
+
+![image](https://user-images.githubusercontent.com/24860547/200813751-b5c9d767-18ab-489c-9157-dfea041a7f12.png)
+
+
+### Statistics
+
+The statistics charts measure lifetime averages - averages of metrics over the lifetime of the Apache server being up and operational.
+
+The first chart shows the lifetime average of requests per second - notice that this is significantly different from the first chart we mentioned (rate of requests). Occasional spikes and dips may not register on this lifetime average chart but it is still very useful for understanding longer term trends of resource utilization.
+
+![image](https://user-images.githubusercontent.com/24860547/200813866-ca6c9161-6474-4be9-9684-2213b142b74c.png)
+
+The next couple of charts show the lifetime average of, bytes served and response size. Both of these metrics can also be useful information to understand how the server is performing for the current use-case and is valuable in terms of designing potential upgrades to the server.
+
+![image](https://user-images.githubusercontent.com/24860547/200813938-144584e5-30ee-4de5-b4a2-4b230b0f9992.png)
+![image](https://user-images.githubusercontent.com/24860547/200814009-ddf8e625-d5f3-4d43-bae9-5da3edb85841.png)
+
+
+### Availability (Uptime)
+
+The uptime of the Apache server is monitored by this chart and helps you quickly get an idea if the server had any downtime.
+
+![image](https://user-images.githubusercontent.com/24860547/200814062-5c2536d4-a264-407f-9686-2d642402334a.png)
+
+
+## Troubleshooting Apache with Netdata
+
+### Alerts
+
+Netdata has a built-in health watchdog that comes with pre configured alerts to reduce the monitoring burden for you. 
```

Comment:
```suggestion
Netdata has a built-in health engine that comes with pre configured alerts to reduce the monitoring burden for you. 
```

Reply:
"watchdog" is a datadog product feature. Also engine sounds better i think. 

...actually wait - looks like we using watchdog here too: https://learn.netdata.cloud/docs/agent/health/

when we start using the "health watchdog" term? is a bit too marketing buzzy imo. 

"health engine" is how i know it to be called. 

anyway just ranting here - i'd have pushed based on just deciding to use the "watchdog" term and its one of those fluffy meaningless terms we tend to try not use.

Reply:
@DShreve2 ping since i'm ranting :)  - where did "health watchdog" term come from?

Reply:
we've been using this one for a while, any idea @sashwathn ?

Reply:
Though, "watchdog" is not necessarily a marketing term - I think linux users generally understand it to mean a daemon that monitors if some condition has been met to then trigger some action.

Reply:
Indeed health engine is probably a better term and is used in a number of places and the module itself is called the "health engine".
We could replace watchdog with engine in all our docs just for the sake of uniformity.